packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -102,8 +102,6 @@ const FlowErrors = Object.freeze({
   SpreadVariance: "Spread properties cannot have variance",
   ThisParamAnnotationRequired:
     "A type annotation is required for the `this` parameter.",
-  ThisParamBannedInArrowFunctions:
-    "Arrow functions cannot have a `this` parameter; arrow functions automatically bind `this` when declared.",
```

Comment:
`this` is not allowed within params of arrow functions: https://flow.org/try/#0BQFwFglgzgXABAMQJRwLwD44G8C+Q

Reply:
yes, this is already an error in babel, there is a test for this